src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -62,25 +64,16 @@ public String getName() {
   }
 
   @Override
-  @NotNull
+  @Nullable
   public HaxeComponentName getComponentName() {
-    return new HaxeComponentNameImpl(getNode());
-    ////this.getName();
-    //return findNotNullChildByClass(HaxeComponentName.class);
+    return null;
```

Comment:
To reproduce - just type `var a = 0;` and move caret back `var a = <caret>0;` and for LocalVarDeclaration getIdentifier will be required from getComponentName.
```
PsiIdentifier id = getNameIdentifier();
return null != id ? new HaxeComponenetNameImp(id.getNode()) : null;
```
Will cause endless recursion. Notice that `getNameIdentifier` is declared for Mixin